Modules/main.c

```diff
@@ -1828,6 +1834,23 @@ pymain_main(_PyMain *pymain)
         }
     }
 
+    if (_Py_UnhandledKeyboardInterrupt) {
```

Comment:
Why are you doing this before `pymain_free(pymain);`

Reply:
Would no be good make a ```pymain_free(pymain)``` when you ```kill()``` ?

Reply:
Agreed, it belongs below.  moved.  `pymain_free` doesn't actually deallocate pymain itself despite the name "free" which is also why the existing code was okay doing `return pymain->status;`.